program-test/src/lib.rs

```diff
@@ -698,3 +700,62 @@ impl ProgramTestBanksClientExt for BanksClient {
         ))
     }
 }
+
+struct DroppableTask<T>(Sender<()>, JoinHandle<T>);
+
+impl<T> Drop for DroppableTask<T> {
+    fn drop(&mut self) {
+        match self.0.try_send(()) {
+            Ok(_) => {}
+            Err(e) => {
+                debug!("Error cancelling task: {:?}", e);
+            }
+        }
+    }
+}
+
+pub struct ProgramTestState {
```

Comment:
nit: I think I fancy "context" more:
```suggestion
pub struct ProgramTestContext {
```

Reply:
Works for me!